cpu/stm32/periph/adc_wl.c

```diff
@@ -152,8 +207,13 @@ int32_t adc_sample(adc_t line, adc_res_t res)
         vbat_disable();
     }
 
-    /* free the device again */
+    /* disable, unlock and power off the device again */
+    int ret = _disable_adc();
     done();
 
-    return sample;
+    if (ret == -1) {
+        return -1;
+    } else {
```

Comment:
```suggestion
    }
    else {
```

The coding convention prefers this ;)